packages/agents/chain-abstraction/src/helpers/swaputil.ts

```diff
@@ -0,0 +1,107 @@
+import { AlphaRouter } from "@uniswap/smart-order-router";
+import { ethers } from "ethers";
+import { CurrencyAmount, Token, TradeType } from "@uniswap/sdk-core";
+import { Route, Trade, Token as _Token, Fetcher, TokenAmount, Pair } from "@uniswap/sdk";
+// import { Route, Trade, TokenAmount, Fetcher, Token, TradeType, CoWTrade } from "@swapr/sdk";
+import {
+  Fetcher as PancakeFetcher,
+  Trade as PancakeTrade,
+  Token as PancakeToken,
+  Route as PancakeRoute,
+  CurrencyAmount as PancakeCurrencyAmount,
+} from "@pancakeswap/sdk";
+
+type SwapPathCallBackArgs = {
+  fromTokenContractAddress: string;
+  toTokenContractAddress: string;
+  signerAddress: string;
+  chainId: number;
+  rpc: string;
+  amount: string;
+  fromTokenDecimal?: number;
+  toTokenDecimal?: number;
+};
+
+export type DestinationSwapPathCallback = (args: SwapPathCallBackArgs) => Promise<any>;
+
+/**
+ * Returns the `path` from the uniswap v3 router.
+ */
+export const getSwapPathForUniV3 = async (_args: SwapPathCallBackArgs) => {
+  try {
+    const { fromTokenContractAddress, toTokenContractAddress, chainId, rpc, fromTokenDecimal, toTokenDecimal, amount } =
+      _args;
+    const provider = new ethers.providers.JsonRpcProvider(rpc);
+    const tokenIn = new Token(chainId, fromTokenContractAddress, fromTokenDecimal ?? 18);
+    const tokenOut = new Token(chainId, toTokenContractAddress, toTokenDecimal ?? 18);
+    const amountIn = CurrencyAmount.fromRawAmount(tokenIn, +amount);
+
+    const router = new AlphaRouter({
+      chainId,
+      provider,
+    });
+
+    const routes = await router.route(amountIn, tokenOut, TradeType.EXACT_INPUT);
+
+    if (!routes) {
+      throw Error("No routes found for given Pair");
+    }
+    return {
+      quote: routes.quote,
+      tokenPath: routes.route[0].tokenPath.map((token) => token.address),
```

Comment:
UniV3 pair has fee percent, that is different with univ2 pair. So path should be `token-fee-token(bytes)` array.

https://github.com/Uniswap/v3-periphery/blob/6cce88e63e176af1ddb6cc56e029110289622317/contracts/libraries/Path.sol#L42